### PR TITLE
Add method to control the sys info cache

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
If the module is used in long running processes we must provide the users a way to disable the cache for all checks in sys info.

### CODE REVIEW

File: lib/helpers.ts [hunk 1]
```diff
@@ -1,6 +1,17 @@
 import { platform } from "os";
 
 export class Helpers {
+	public getPropertyName(method: Function): string {
+		if (method) {
+			let match = method.toString().match(/(?:return\s+?.*\.(.+);)|(?:=>\s*?.*\.(.+)\b)/);
```
Comment (rosen-vladimirov) [inline lib/helpers.ts]: this looks like an absolute magic, but I'll believe you that it works

File: lib/sys-info.ts [hunk 2]
```diff
@@ -311,26 +291,43 @@ export class SysInfo implements NativeScriptDoctor.ISysInfo {
...
+		this.shouldCache = shouldCache;
+	}
+
+	private async getValueForProperty<T>(property: Function, getValueMethod: () => Promise<T>): Promise<T> {
+		if (this.shouldCache) {
+			const propertyName = this.helpers.getPropertyName(property);
+			const cachedValue: T = (<any>this)[propertyName];
 
-		return this.isCocoaPodsUpdateRequiredCache;
+			if (cachedValue === undefined || cachedValue === null) {
```
Comment (rosen-vladimirov) [inline lib/sys-info.ts]: most of the methods that should be cached, return `null`, for example `getJavaCompilerVersion`. Does it mean that if I have a problem with Javac, each execution will not  be cached?

**Comment by TsvetanMilanov** [comment]
run ci

**Comment by TsvetanMilanov** [comment]
run ci

**Pull request merged**